### PR TITLE
Finishing models and creating database

### PR DESCRIPTION
**Pull request by duykhang91**

### CODE REVIEW

File: InternProject/DAL/Models/DeliveryTrack/DCBookingModel.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        [StringLength(12)]
+        public string BookingTime { get; set; }
+
+        [StringLength(30)]
+        public string Haulier { get; set; }
+
+        [StringLength(30)]
+        public string CreatedBy { get; set; }
+
+        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:MM/dd/yyyy}")]
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/DeliveryTrack/DCBookingModel.cs]: This is data annotation attribute for ASP.NET MVC only, no effect for EF

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
+using InternProject.DAL.Models.OrderTrack;
+using InternProject.DAL.Models.ShipmentTrack;
+
+namespace InternProject.DAL.Models.DeliveryTrack
+{
+    public class DCBookingDetailModel
+    {
+        [Key]
+        public int Id { get; set; }
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs:15]: Create a base model for all entity to include all common fields such as Id

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        public string Container { get; set; }
+
+        [ForeignKey("DCBookingModel")]
+        public int DCBookingId { get; set; }
+
+        public virtual OrderModel Order { get; set; }
+
+        //public virtual ManifestModel ManifestModel  { get; set; }
+
+        public virtual DCBookingModel DCBookingModel { get; set; }
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs]: Change property name to DCBooking

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        [StringLength(50)]
+        public string Item { get; set; }
+
+        [ForeignKey("Order")]
+        public int OrderId { get; set; }
+
+        //[ForeignKey("ManifestModel")]
+        public string Container { get; set; }
+
+        [ForeignKey("DCBookingModel")]
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs]: For the default behavior of EF, DCBookingId and DCBooking property are naming convention which will automatically  setup a foreign key

Don't need to specify [ForeignKey] here, after you change to DCBooking below

File: InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace InternProject.DAL.Models.OrderTrack
+{
+    public class OrderDetailModel
+    {
+        [Key]
+        [Required]
+        [Display(Name = "Item Number")]
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs]: The same, this is only for MVC, not EF

File: InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        [Range(0, 10 ^ 11 - 1)] //up to 10 digits
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs]: and here

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+namespace InternProject.DAL.Models.OrderTrack
+{
+    public class ProgressCheckModel
+    {
+        [Key]
+        [Required]
+        public int ID { get; set; }
+
+        //yes/no
+        public string Complete { get; set; }
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: how about boolean?

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        public DateTime IntendedShipDate { get; set; }
+
+        // = sum of all revise quantity?
+        public int EstQtyToShip { get; set; }
+
+        //calendar control
+        public DateTime InspectionDate { get; set; }
+
+        //max string length
+        public string COMMENTS { get; set; }
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: naming convention

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        [Key]
+        [Required]
+        public int ID { get; set; }
+
+        //yes/no
+        public string Complete { get; set; }
+
+        //yes/no
+        public string onSchedule { get; set; }
```
Comment (mrkhangtran) [inline InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: naming convention

File: InternProject/DAL/Models/DeliveryTrack/DCBookingModel.cs (hunk 1 above)
Comment (duykhang91) [reply InternProject/DAL/Models/DeliveryTrack/DCBookingModel.cs]: remove data annotations that don't affect EF

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs (hunk 2 above)
Comment (duykhang91) [reply InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs:15]: change field names to match naming convention ("ID" to "Id")

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs (hunk 3 above)
Comment (duykhang91) [reply InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs]: change property name (DCBookingModel to DCBooking, etc.)

File: InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs (hunk 4 above)
Comment (duykhang91) [reply InternProject/DAL/Models/DeliveryTrack/DCBookingDetailModel.cs]: commented out unecessary [ForeignKey] annotation

File: InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs (hunk 5 above)
Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs]: removed [Display] annotation

File: InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs (hunk 6 above)
Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/OrderDetailModel.cs]: removed [Display] annotation

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs (hunk 7 above)
Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: change from "string" to "bool"

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs (hunk 8 above)
Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: change from "COMMENTS" to "Comments"

File: InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs (hunk 9 above)
Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: change from "onSchedule" to "OnSchedule"
change from "string" to "bool"

Comment (duykhang91) [reply InternProject/DAL/Models/OrderTrack/ProgressCheckModel.cs]: change from "string" to "bool"

**Review by mrkhangtran — APPROVED** [review]
architecture to be fixed later

**Pull request merged**